sim/visuals/microbit.ts

```diff
@@ -2552,152 +2566,211 @@ namespace pxsim.visuals {
             }, false);
         }
 
-        private attachPinsIOEvents() {
-            this.pins.slice(0, 4).forEach((pin, index) => {
-                // var index = i + 2;
-                if (!this.board.edgeConnectorState.pins[index]) return;
+        private attachPinsTouchEvents() {
+            const pins = this.pins.slice(0, 5);
+
+            for (let index = 0; index < pins.length; index++) {
+                const pin = pins[index];
+
+                const boundingBox = (pin as SVGPathElement).getBBox();
+
+                const eventSurface = svg.child(
+                    this.buttonGroup,
+                    "rect",
+                    {
+                        "class": "button-rect",
+                        x: boundingBox.x,
+                        y: boundingBox.y - 10,
+                        width: boundingBox.width,
+                        height: boundingBox.height + 20
+                    }
+                );
+
+                svg.title(eventSurface, pin.firstElementChild.textContent);
+
+                let state = this.board;
+                let pressedTime: number;
+
+                for (const eventName of pointerEvents.down) {
+                    eventSurface.addEventListener(eventName, () => {
+                        // console.log(`down ${state.edgeConnectorState.pins[i].id}`)
```

Comment:
Do we still want to keep this in? (applies to all the commented out console logs)

Reply:
i think these were added by the calliope folks so i just left them alone